src/common/args.cpp

```diff
@@ -0,0 +1,825 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <common/args.h>
+
+#include <chainparamsbase.h>
+#include <logging.h>
+#include <sync.h>
+#include <tinyformat.h>
+#include <util/check.h>
+#include <util/fs.h>
+#include <util/fs_helpers.h>
+#include <util/settings.h>
+#include <util/strencodings.h>
+
+#ifndef WIN32
+#include <algorithm>
+#include <cassert>
```

Comment:
In commit "move-only: Extract common/args and common/config.cpp from util/system" (ec51c252de42e11906f114ac05e476fd88ca2176)

Any reason these header are only included if WIN32 is not defined? I know the ifndef is preexisting, but it seems like it would make more sense to make these includes unconditional.

Reply:
I checked it and its history again and can't think of a good reason either. Should I amend here, or go for the follow up?

Reply:
> I checked it and its history again and can't think of a good reason either. Should I amend here, or go for the follow up?

Either way seems fine. Would be nice to turn these into simple includes, but I was asking mostly out of curiosity